src/coreComponents/python/modules/pygeosx_tools_package/pygeosx_tools/mesh_interpolation.py

```diff
@@ -0,0 +1,202 @@
+            binned_values[ii] = np.NaN
+
+    return binned_values
+
+
+def extrapolate_nan_values(x, y, slope_scale=0.0):
+    """
+    Fill in any nan values in two 1D arrays by extrapolating
+
+    Args:
+        x (np.ndarray): 1D list/array of positions
+        y (np.ndarray): 1D list/array of values
+        slope_scale (float): value to scale the extrapolation slope (default=0.0)
+        
+    Returns:
+        np.ndarray: The input array with nan values replaced by extrapolated data
+    """
+    Inan = np.isnan(y)
+    reg = stats.linregress(x[~Inan], y[~Inan])
+    y[Inan] = reg[0] * x[Inan] * slope_scale + reg[1]
+    return y
+
+
+def get_random_realization(x, bins, value, rand_fill=0, rand_scale=0, slope_scale=0):
+    """
+    Get a random realization for a noisy signal with a set of bins
+
+    Args:
+        x (np.ndarray): 1D list/array of positions
+        bins (np.ndarray): 1D list/array of bin edges
+        value (np.ndarray): 1D list/array of values
+        rand_fill (float): The standard deviation to use where data is not defined (default=0)
+        rand_scale (float): Value to scale the standard deviation for the realization (default=0)
+        slope_scale (float): Value to scale the extrapolation slope (default=0.0)
+
+    Returns:
+        np.ndarray: An array containing the random realization
+    """
+    y_mean = apply_to_bins(np.mean, x, value, bins)
+    y_std = apply_to_bins(np.std, x, value, bins)
+
+    # Extrapolate to fill the upper/lower bounds
+    x_mid = bins[:-1] + 0.5 * (bins[1] - bins[0])
+    y_mean = extrapolate_nan_values(x_mid, y_mean, slope_scale)
+    y_std[np.isnan(y_std)] = rand_fill
+
+    # Add a random perturbation to the target value to match missing high/lows
+    y_final = y_mean + (rand_scale * y_std * np.random.randn(len(y_mean)))
+    return y_final
+
+
+def get_realizations(x, bins, targets):
+    """
+    Get random realizations for noisy signals on target bins
+
+    Args:
+        x (np.ndarray): 1D list/array of positions
+        bins (np.ndarray): 1D list/array of bin edges
+        targets (dict): Dict of geosx target keys, inputs to get_random_realization
+
+    Returns:
+        dict: Dictionary of random realizations
+
+    """
+    results = {}
+    for ka in targets.keys():
+        results[ka] = get_random_realization(x, bins, **targets[ka])
+    return results
+
+
+def get_surface_interpolator(fname, z_origin=2700.0):
+    """
+    Parse a surface definition file
+
+    Args:
+        fname (str): Path of the target file
+        z_origin (float): Vertical offset of a target mesh file (default=2700.0)
+
+    Returns:
+        tuple: Surface interpolators (linear-depth, nearest-depth, linear-x, nearest-x)
+    """
+    nodes = []
+    with open(fname, 'r') as f:
+        for line in f:
+            if ('#' not in line):
+                tmp = line.split(',')
+                if ('.' in line):
+                    nodes.append([float(x) for x in tmp])
```

Comment:
```suggestion
    with open(fname, 'r') as f:
        for line in filter(lambda l: ('#' not in line) and ('.' in line), f):
            nodes.append([float(x) for x in line.split(',')])
```
is more pythonic (if I did nothing wrong).

Reply:
I've deleted this function, because it is too specific to the problem I was working on.